UML-Diagram/StatisticSystem.puml

```diff
@@ -0,0 +1,29 @@
+@startuml StatisticSystem
+package "Statistic System" {
+    
+    ' Define abstract classes
+    
+    abstract class StatisticReport {
+        - BorrowManager : Borrows
+        + abstract Report()
+    }
+    
+    ' Define main classes
+
+    class MostBorrowedBook extends StatisticReport {
+        - BorrowManager : Borrows
+        + override Report ()
+    }
+
+    class MostActivePatron extends StatisticReport {
+        - BorrowManager : Borrows
+        + override Report ()
+    }
+
+    class Statistic {
+        - StatisticReport : Report
+        + void makeStatistic()
+    }
+
+}
+@enduml 
```

Comment:
You have used the factory design pattern, but I would like you to tell me why it is the best choice and what benefits it will give us for this implementation.

Reply:
I used strategy pattern because its fit with the definition, strategy let us define a family of algorithm in this case statistics but each of them in separate class. This because it allows us to implement more statistics just to implement new classes that implemens the abstract class.